SourceX/qol.cpp

```diff
@@ -216,4 +216,156 @@ void AutoGoldPickup(int pnum)
 	}
 }
 
+struct drawingQueue {
+	int itemID, row, col, x, y, width, height;
+	text_color color;
+	std::string text;
+	drawingQueue(int itemID, int row, int col, int x, int y, int width, int height, text_color color, std::string text)
+	    : itemID(itemID)
+	    , row(row)
+	    , col(col)
+	    , x(x)
+	    , y(y)
+	    , width(width)
+	    , height(height)
+	    , color(color)
+	    , text(text)
+	{
+	}
+};
+
+std::vector<drawingQueue> drawQ;
+int drawMinX;
+int drawMaxX;
+bool altPressed = false;

```

Comment:
It'd be good to separate out the input handling from functionality.
One reason is that for gamepads I think we want this to be a toggle.

Should we rename this to something like `bGroundItemsHighlight`?

Reply:
sure, I just didn't think I'd get this to work today at all :D

Reply:
Making it just "highlight" might be misleading. Highlighting already exists both when you mouse over as well as when you use the search spell.

I think we need to come up with a different name for this behavior.

What about `ShowItemNameTooltips`?

Reply:
> It'd be good to separate out the input handling from functionality.
> One reason is that for gamepads I think we may want this to be a toggle.
> 
> Should we rename this to something like `bGroundItemsHighlight`?

Gamepads can call ToggleItemLabelHighlight() to turn highlighting on/off now